contracts/token/BiconomyTokenPaymaster.sol

```diff
@@ -584,11 +584,22 @@ contract BiconomyTokenPaymaster is
                 priceSource
             );
         } else {
-            //in case above transferFrom failed, pay with deposit / notify at least
-            //sender could be banned indefinitely or for certain period
-            emit TokenPaymentDue(address(feeToken), account, charge);
-            // review
-            // return; // Do nothing here to not revert the whole bundle and harm reputation
+            // In case transferFrom failed in first handlePostOp call, attempt to charge the tokens again
+            bytes memory _data = abi.encodeWithSelector(
+                feeToken.transferFrom.selector,
+                account,
+                feeReceiver,
+                charge
```

Comment:
Should we increase the charge here to cover the gas used for extra logic here or the token markup will cover these as this flow is not expected to happen frequently?

Reply:
we need to recalculate with updated UNACCOUNTED_COST with max execution gas postOp can consume for this case. should this value be another constant fed to the contract?

also I just thought, currently UNACCOUNTED_COST is settable but it's fixed value for specific token where in reality - dror agreed  on infinitism PR - postop_cost depends on price of token transfer, which varies with different token implementation. 

It's difficult to do this in current design but only premium markups to adjust it still.